taipy/core/data/data_node.py

```diff
@@ -415,19 +420,19 @@ def read(self) -> Any:
             )
             return None
 
-    def append(self, data, job_id: Optional[JobId] = None, **kwargs: Dict[str, Any]):
+    def append(self, data, editor_id: Optional[str] = None, **kwargs: Dict[str, Any]):
         """Append some data to this data node.
 
         Arguments:
             data (Any): The data to write to this data node.
-            job_id (JobId): An optional identifier of the writer.
+            editor_id (str): An optional identifier of the editor.
             **kwargs (dict[str, any]): Extra information to attach to the edit document
                 corresponding to this write.
         """
         from ._data_manager_factory import _DataManagerFactory
 
         self._append(data)
-        self.track_edit(job_id=job_id, **kwargs)
+        self.track_edit(editor_id=editor_id, **kwargs)
```

Comment:
will the current editor_id (set by lock_edit) be picked up automagically or is it not needed ?

Reply:
I don't understand the question. What do you mean by: "be picked up automagically or is it not needed" ?

The `editor_id` must be explicitly passed by the client when writing/appending the data so we can ensure it is the right editor and not someone else. The track_edit method is automatically called by the write and append methods with the appropriate editor id.

Reply:
Oh, ok. I get it now. This PR is really about cleaning the parameters of the track_edit method. But indeed, it is related to the behavior of the lock_edit and write methods.

Yes, we should change that in the future. There will be another (hacktoberfest) PR that will simplify the API soon, so you won't need to call the unlock_edit after writing the data.

In any case, what you did is working just fine because you ensure there is no concurrent edition on your side, and when you don't pass any editor_id to the write function, the data is still written. No problem.

I let you know when the other PR is done as well.